modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/services/ApiHelper.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {fetch} from 'frontend-js-web';
+
+const HEADERS = new Headers({
+	'Accept': 'application/json',
+	'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
+	'Content-Type': 'application/json',
+});
+
+export const HEADERS_ALL_LANGUAGES = new Headers({
+	'Accept': 'application/json',
+	'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
+	'Content-Type': 'application/json',
+	'X-Accept-All-Languages': 'true',
+});
+
+const UNEXPECTED_ERROR_MESSAGE = Liferay.Language.get(
+	'an-unexpected-error-occurred'
+);
+
+type RequestResult<T> =
+	| {
+			data: null;
+			error: string;
+	  }
+	| {
+			data: T;
+			error: null;
+	  };
+
+async function handleRequest<T>(
+	fetcher: () => Promise<Response>
+): Promise<RequestResult<T>> {
+	try {
+		const response = await fetcher();
+
+		if (response.status === 401) {
+			window.location.reload();
+		}
+
+		if (!response.ok) {
+			let errorMessage = UNEXPECTED_ERROR_MESSAGE;
```

Comment:
Could we declare this variable on line 50?

Reply:
Yep! I added [a commit](https://github.com/liferay-content-management/liferay-portal/pull/6645/commits/d5512f87282a8c55b93eebb9321b7a5fae1eb2f9) improving that part, thanks!

Reply:
I don't have permissions on Sandro's remote, so I will forward this pull, and make this change in my next task https://liferay.atlassian.net/browse/LPD-55202